usaspending_api/download/management/commands/populate_monthly_delta_files.py

```diff
@@ -0,0 +1,189 @@
+import logging
+import os
+import re
+import shutil
+import subprocess
+import tempfile
+import zipfile
+
+from collections import OrderedDict
+from datetime import datetime, date
+from django.conf import settings
+from django.core.management.base import BaseCommand
+from django.db.models import Case, When, Value, CharField
+
+from usaspending_api.awards.v2.lookups.lookups import all_award_types_mappings
+from usaspending_api.common.helpers import generate_raw_quoted_query
+from usaspending_api.download.filestreaming.csv_generation import EXCEL_ROW_LIMIT, CsvSource
+from usaspending_api.download.helpers import split_csv, pull_modified_agencies_cgacs, multipart_upload
+from usaspending_api.download.lookups import VALUE_MAPPINGS
+from usaspending_api.references.models import ToptierAgency
+
+logger = logging.getLogger('console')
+
+AWARD_MAPPINGS = {
+    'Contracts': {
+        'award_types': ['contracts'],
+        'correction_delete_ind': 'correction_delete_ind',
+        'date_filter': 'updated_at',
+        'letter_name': 'd1',
+        'model': 'contract_data'
+    },
+    'Assistance': {
+        'award_types': ['grants', 'direct_payments', 'loans', 'other_financial_assistance'],
+        'correction_delete_ind': 'transaction__assistance_data__correction_late_delete_ind',
+        'date_filter': 'modified_at',
+        'letter_name': 'd2',
+        'model': 'assistance_data'
+    }
+}
+
+
+class Command(BaseCommand):
+
+    def download(self, award_type, agency='all', generate_since=None):
+        """Create a delta file based on award_type, and agency_code (or all agencies)"""
+        award_map = AWARD_MAPPINGS[award_type]
+
+        # Create Source and update fields to include correction_delete_ind
+        source = CsvSource('transaction', award_map['letter_name'].lower(), 'transactions')
+        source.query_paths.update({
+            'correction_delete_ind': award_map['correction_delete_ind']
+        })
+        source.query_paths.move_to_end('correction_delete_ind', last=False)
+        source.human_names = list(source.query_paths.keys())
+
+        # Apply filters to the queryset
+        filters, agency_code = self.parse_filters(award_map['award_types'], agency)
+        source.queryset = VALUE_MAPPINGS['transactions']['filter_function'](filters)
+        if award_type == 'Contracts':
+            # Derive the correction_delete_ind from the created_at of the records
```

Comment:
Stupid question: why is this being applied to contracts? I thought both the `correction_delete_ind` were being added via line 50-52?

Reply:
The `correction_delete_ind` in the `AWARD_MAPPINGS` represents the mapping from query to file header, with the key being the file-level header and the value being the column name within the query.

The code here defines the data within the query in order for it to be mapped to a file header in the first place.